### DISCUSSION TITLE
New API Documentation - Developer Preview

### DISCUSSION THREAD

**Discussion by MYoung25** [aws/aws-sdk-js-v3#4656] [General]
### We are excited to announce the [developer preview](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/preview/) of our new API documentation for AWS SDK for JavaScript v3.

This discussion is for all things related to our new documentation.  Please feel free to provide feedback here or open an issue.

#### Key Improvements:
- Syntax examples and example usage
- Simplified navigation and discoverability via global search
- Ability to provide feedback and suggest edits for specific pages
- Reduction of implementation details and internal types being displayed prominently

#### Outstanding Issues:
- [x] Client Configuration options are not exposing nested type information
- [x] Packages and Libraries have not been migrated yet, see [coming soon](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/preview/coming-soon/)
- [x] Global Search only supports Service and Command names
- [x] Some nested type links show a blank help panel and/or lead to a 404 page

**Comment by rehanvdm** [comment]
Love the example usage section that has a drop-down. On this outstanding issue:
> Search only supports Service and Command names

I thought that this might be a feature, because if you are on the [S3 page](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/preview/client/s3/) and search for `put object`, it would match multiple commands that just mention it, which isn't what I would expect, I expect to see the commands with that text. So maybe have a check box that if checked also searches the command descriptions? 

A few other things: 
1. How about moving the XXX Client Configuration section down below the search bar, or just moving the search bar up. I want to start searching as soon as I open the page
2. How about not ordering the Input Parameters alphabetically, but rather show the required ones first? I always find myself scrolling to find them first. Or maybe another check box next to the Input Parameters (show only required ) or something like that.

**Reply by MYoung25** [reply]
Thanks for leaving feedback!

Apologies for the confusion on search, there are multiple places to "search" in the new docs.  I've updated the conversation to clarify that only the _Global_ search (in the top bar) is not intended to be limited to Client and Command names.

On a client's page, you are correct that the intended behavior for filtering is _only_ on command names.

**Comment by rehanvdm** [comment]
The example section is editable code. I am not sure if it is intentional or not, but if it is, it would be nice to have autocomplete work there 
![image](https://user-images.githubusercontent.com/22810856/234166543-aff65fc2-9620-4bcf-b2e1-a65931563a0c.png)

**Comment by MYoung25** [comment]
Requests to include example responses in the `Example Syntax`

[radzikowski_m](https://twitter.com/radzikowski_m/status/1650799746544435201)
> no Command response interface description or link? I understand all input params are in the example code, but there is no response format given.

[jesterxl](https://twitter.com/jesterxl/status/1650835663170809860?s=20)
> love the "single page for the Command". However, missing the response example (see line 46). Line 4 shows the input  example == awesome, but we need the response too.
https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/preview/client/s3/command/PutObjectCommand/
It's on line 13 in the 2nd example, but I'm already doing a ton of scrolling.

**Reply by MYoung25** [reply]
Another data point to support this: #4662

**Reply by MYoung25** [reply]
I'm happy to report that Example Syntax now includes the output shape!

**Reply by cdimino** [reply]
+1 to seeing output on [this page](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/preview/client/ssm/command/PutParameterCommand/), like in [in current docs](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-ssm/interfaces/putparametercommandoutput.html).  Or is it there?  I might be blind.

**Comment by pranjalja1n** [comment]
The type description sidebar is of a fixed width and truncates information, better UX would be to either open the sidebar to it's full width so that content is not truncated or make it re-size able.

![image](https://github.com/aws/aws-sdk-js-v3/assets/24854511/36182595-d3a9-4ad2-a072-bd125fc15ad4)

**Reply by MYoung25** [reply]
Thanks for taking time to leave us feedback! I agree that the right sidebar is difficult to use in its current format.  Initially it wasn't meant to have quite so much information in it, and clearly its grown a bit.  I've brought this up to the team and have started thinking about some creative ways to make it easier to digest.

Best,

Michael
